File: src/main/java/uk/edu/populationfitness/models/genes/bitset/InvertedBitSetGenes.java
package uk.edu.populationfitness.models.genes.bitset;

import uk.edu.populationfitness.models.Config;

abstract class InvertedBitSetGenes extends BitSetGenes {
    InvertedBitSetGenes(Config config) {
        super(config);
    }

    /**
     * Call this to store the fitness, inverted to 1- fitness
     *
     * @param fitness
     * @return the scaled stored fitness
     */
    double storeInvertedFitness(double fitness){
        return storeFitness(1 - fitness);
    }
}

File: DotNet/PopulationFitness/PopulationFitness/Models/Genes/BitSet/InvertedBitSetGenes.cs
namespace PopulationFitness.Models.Genes.BitSet
{
    public abstract class InvertedBitSetGenes : BitSetGenes
    {
        protected InvertedBitSetGenes(Config config) : base(config)
        {
        }

        /**
         * Call this to store the fitness, inverted to 1- fitness
         *
         * @param fitness
         * @return the scaled stored fitness
         */
        protected double StoreInvertedFitness(double fitness)
        {
            return StoreFitness(1 - fitness);
        }
    }
}
